src/Readers/ODataReader.v4/ODataCapabilitiesReader.cs

```diff
@@ -0,0 +1,196 @@
+            odcmProperty.Projection.EnsureCapabilities().Insertable = insertable;
+
+            foreach (var propertyPath in nonInsertableNavigationProperties)
+            {
+                OdcmProperty navProperty;
+                if (!(odcmProperty.Type as OdcmClass).TryFindProperty(propertyPath, out navProperty))
+                {
+                    throw new InvalidOperationException();
+                }
+                navProperty.Projection.EnsureCapabilities().Insertable = false;
+            }
+        }
+
+        private static void GetUpdateRestrictions(OdcmProperty odcmProperty, IEdmModel model, IEdmEntitySet entitySet)
+        {
+            IEdmValueAnnotation updateAnnotation = FindVocabularyAnnotation(model, entitySet, s_updateRestrictionsTerm);
+            if (updateAnnotation == null)
+            {
+                return;
+            }
+
+            bool updatable;
+            IEnumerable<string> nonupdatableNavigationProperties;
+            GetBooleanAndPathCollection(updateAnnotation, "Updatable", "NonUpdatableNavigationProperties", out updatable, out nonupdatableNavigationProperties);
+            odcmProperty.Projection.EnsureCapabilities().Updateable = updatable;
+
+            foreach (var propertyPath in nonupdatableNavigationProperties)
+            {
+                OdcmProperty navProperty;
+                if (!(odcmProperty.Type as OdcmClass).TryFindProperty(propertyPath, out navProperty))
+                {
+                    throw new InvalidOperationException();
+                }
+                navProperty.Projection.EnsureCapabilities().Updateable = false;
+            }
+        }
+
+        private static void GetDeleteRestrictions(OdcmProperty odcmProperty, IEdmModel model, IEdmEntitySet entitySet)
+        {
+            IEdmValueAnnotation deleteAnnotation = FindVocabularyAnnotation(model, entitySet, s_deleteRestrictionsTerm);
+            if (deleteAnnotation == null)
+            {
+                return;
+            }
+
+            bool deletable;
+            IEnumerable<string> nonDeletableNavigationProperties;
+            GetBooleanAndPathCollection(deleteAnnotation, "Deletable", "NonDeletableNavigationProperties", out deletable, out nonDeletableNavigationProperties);
+            odcmProperty.Projection.EnsureCapabilities().Deleteable = deletable;
+
+            foreach (var propertyPath in nonDeletableNavigationProperties)
+            {
+                OdcmProperty navProperty;
+                if (!(odcmProperty.Type as OdcmClass).TryFindProperty(propertyPath, out navProperty))
+                {
+                    throw new InvalidOperationException();
+                }
+                navProperty.Projection.EnsureCapabilities().Deleteable = false;
+            }
+        }
+
+        private static void GetExpandRestrictions(OdcmProperty odcmProperty, IEdmModel model, IEdmEntitySet entitySet)
```

Comment:
The name here is confusing because the method does not return a value. Consider modeling this as an extension method on odcmProperty, showing who is being acted upon, and rename it to something like SetNavigationProperty*Restrictions